### PR TITLE
Capture all files referenced in result to files collection. Verify we…

### PR DESCRIPTION
**Pull request by michaelcfanning**
… do not capture files referenced in tool notifications. Update result level default to 'Default' rather than 'Unknown' (because the default value does not indicate something unknown but an actual default, such as Warning).

@lgolding @rtaket

### CODE REVIEW

File: src/Sarif.Converters/FxCopConverter.cs [hunk 1]
```diff
@@ -197,7 +197,7 @@ private static ResultLevel ConvertFxCopLevelToResultLevel(string fxcopLevel, out
             // FxCop provides no MessageLevel. For these issues, we shouldn't
```
Comment (lgolding) [inline src/Sarif.Converters/FxCopConverter.cs]: The comment now doesn't quite match the code.

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 2]
```diff
@@ -153,6 +155,128 @@ public void SarifLogger_WritesFileData()
...
+                                Frames = new[]
+                                {
+                                    new StackFrame
+                                    {
+                                        Uri = new Uri(@"file:///file4.cpp")
+                                    }
+                                }
+                            }
+                        }
+                    };
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: Missing code flows.

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 3]
```diff
@@ -153,6 +155,128 @@ public void SarifLogger_WritesFileData()
...
+            string logText = sb.ToString();
+            var sarifLog = JsonConvert.DeserializeObject<SarifLog>(logText);
+
+            int fileCount = 5;
+
+            for (int i = 0; i < fileCount; ++i)
+            {
+                string fileName = @"file" + i + ".cpp";
+                string fileDataKey = new Uri("file:///" + fileName).ToString();
+                sarifLog.Runs[0].Files.ContainsKey(fileDataKey).Should().BeTrue("file data for " + fileName + " should exist in files collection");
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: `sarifLog.Runs[0].Files.Should().ContainKey(fileDataKey, "...")`

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 4]
```diff
@@ -153,6 +155,128 @@ public void SarifLogger_WritesFileData()
...
+                    sarifLogger.LogConfigurationNotification(configurationNotification);
+
+                }
+            }
+
+            string logText = sb.ToString();
+            var sarifLog = JsonConvert.DeserializeObject<SarifLog>(logText);
+
+            sarifLog.Runs[0].Files.Count.Should().Be(0);
+        }
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: Or `sarifLog.Runs[0].Files.Should().BeEmpty()`

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 5]
```diff
@@ -153,6 +155,128 @@ public void SarifLogger_WritesFileData()
...
+
+        [TestMethod]
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: Hate when that happens.

**Comment by lgolding** [comment]
:+1:

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs (hunk 2 above)
Comment (michaelcfanning) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: nice catch!

File: src/Sarif.UnitTests/Writers/SarifLoggerTests.cs [hunk 6]
```diff
@@ -273,7 +273,7 @@ public void SarifLogger_DoNotScrapeFilesFromNotifications()
             string logText = sb.ToString();
             var sarifLog = JsonConvert.DeserializeObject<SarifLog>(logText);
 
-            sarifLog.Runs[0].Files.Count.Should().Be(0);
+            sarifLog.Runs[0].Files.Should().BeNull();
```
Comment (lgolding) [inline src/Sarif.UnitTests/Writers/SarifLoggerTests.cs]: Oh, ok. +1

**Pull request merged**